### PR TITLE
DVCSMP-1943 Adding Scenes to Amazon Alexa via Smart Home API

### PR DESCRIPTION
**Pull request by larsfinander**
- Handle restricted routines:
- Send push notification first time routine is executed if restriced
- Educate users about restrictions on OAUTH, Alexa app and device selection page
-  Hack to fix "I'am Back!"

### CODE REVIEW

**Comment by larsfinander** [comment]
@surfous have a look, some text is repeated for debug purposes until alexa decides on which one to go with so this is not final

File: src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy [hunk 1]
```diff
@@ -124,7 +131,8 @@ preferences(oauthPage: "oauthPage") {
 			paragraph title: "Or choose individual devices below", ""
 			input "switches", "capability.switch", title: "My Switches", multiple: true, required: false
 			input "thermostats", "capability.thermostat", title: "My Thermostats", multiple: true, required: false
-			input "routinesEnabled", type: "enum", title: "Enable Routines?", options: [[(true): "Yes"]], defaultValue: false, multiple: false, required: false
+			paragraph title: "Note, executing routines via Alexa will not affect SmartHomeMonitor, locks or garagedoors)", "ddsfdsfdsfdsfdsfds"
```
Comment (surfous) [inline src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy]: trailing ')' after garage doors - not sure if entire string should be in parenthesis, or if this instance should not have parenthesis.

File: src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy [hunk 2]
```diff
@@ -527,6 +564,17 @@ def onOffCommand(device, turnOn, response, routine = null) {
 			}
 		}
 	} else if (routine) {
+			if (routine.hasSecureActions) {
```
Comment (surfous) [inline src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy]: Looks like indents from 567-573 are messed up a bit

File: src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy [hunk 3]
```diff
@@ -527,6 +564,17 @@ def onOffCommand(device, turnOn, response, routine = null) {
...
+			if (state.routineNotifications == null) {
+					state.routineNotifications = [:]
```
Comment (surfous) [inline src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy]: Indents 569-573

File: src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy [hunk 4]
```diff
@@ -190,7 +198,30 @@ def deviceAuthorization() {
 			section("Select Devices") {
 				input "switches", "capability.switch", multiple: true, required: false, title: "Selected Switches"
 				input "thermostats", "capability.thermostat", title: "Selected Thermostats", multiple: true, required: false
-				input "routinesEnabled", "bool", title: "Routines", options: ["All routines"], description: "Select routines", required: false
+				input "routinesEnabled", "bool", title: "Routines", options: ["All routines"], description: "", required: false, submitOnChange:true
+			}
+		}
+		if (isBlanketAuthorized() || areRoutinesEnabled()) {
+				def routines = location.helloHome?.getPhrases()
```
Comment (surfous) [inline src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy:199]: Looks like just additional tab here and in a couple other places in this chunk - lines:
- 205
- 213
- 215
- 221 through 223

**Comment by surfous** [comment]
@larsfinander: Looks good in general - just a few small issues with indents and a rogue close parenthesis in a string, called out in line comments.

File: src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy (hunk 1 above)
Comment (larsfinander) [inline src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy]: Done

File: src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy (hunk 4 above)
Comment (larsfinander) [inline src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy:199]: Done

File: src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy (hunk 3 above)
Comment (larsfinander) [inline src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy]: DOne

**Comment by surfous** [comment]
👍

**Comment by surfous** [comment]
@larsfinander : Tiny change - just capitalize the R in Routines in the definition description (line 13) to match the others as it looks like we standardized capitalizing the R. 

Otherwise 👍  and you can go ahead and merge

**Pull request merged**